aiidalab_widgets_base/computational_resources.py

```diff
@@ -1,23 +1,27 @@
+from __future__ import annotations
+
+import copy
 import enum
 import os
 import subprocess
 import threading
-from copy import copy
+from collections import namedtuple
 from pathlib import Path
 from uuid import UUID
 
 import ipywidgets as ipw
 import pexpect
 import shortuuid
-import traitlets
+import traitlets as tl
 from aiida import common, orm, plugins
 from aiida.orm.utils.builders.computer import ComputerBuilder
 from aiida.transports.plugins import ssh as aiida_ssh_plugin
 from humanfriendly import InvalidSize, parse_size
 from IPython.display import clear_output, display
+from jinja2 import Environment, meta
```

Comment:
The `meta` is not a module so I cannot do `jinja.meta.find_undeclared_variables`. But the `meta` itself definitely very confusing in the context because we have `metadata` for the template. I changed it to:

```python
import jinja2
from jinja2 import meta as jinja2_meta
```